lib/facility.rb

```diff
@@ -1,14 +1,38 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
-
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  attr_reader :name,
+              :address,
+              :phone,
+              :services,
+              :registered_vehicles,
+              :collect_fees
+              
+  def initialize(facility)
+    @name = facility[:name]
+    @address = facility[:address]
+    @phone = facility[:phone]
     @services = []
-  end
+    @registered_vehicles = []
+    @collect_fees = 0
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def register_vehicle(vehicle)
+    if @servies.include('Vehicle Registration')
+      collect_fees(vehicle)
+      vehicle.registration_date == Date.today
+      plate_type(vehicle)
+      @registered_vehicles << vehicle
+  end
+
+  def collect_fees
```

Comment:
Your method here is not taking any args but is when you call it in your above method.